sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Rules/RedundantToCharArrayCall.cs

```diff
@@ -47,21 +47,22 @@ protected override void Initialize(SonarAnalysisContext context)
                 {
                     var invocation = (InvocationExpressionSyntax)c.Node;
 
-                    if (!(c.SemanticModel.GetSymbolInfo(invocation).Symbol is IMethodSymbol methodSymbol) ||
-                        methodSymbol.Name != "ToCharArray" ||
-                        !methodSymbol.IsInType(KnownType.System_String) ||
-                        methodSymbol.Parameters.Length != 0)
+                    if (!(invocation.Parent is ElementAccessExpressionSyntax) &&
+                        !(invocation.Parent is ForEachStatementSyntax))
                     {
                         return;
                     }
 
-                    if (!(invocation.Parent is ElementAccessExpressionSyntax) &&
-                        !(invocation.Parent is ForEachStatementSyntax))
+                    if (!(invocation.Expression is MemberAccessExpressionSyntax memberAccess))
                     {
                         return;
                     }
 
-                    if (!(invocation.Expression is MemberAccessExpressionSyntax memberAccess))
+                    if (!invocation.ToString().Contains("ToCharArray") ||
```

Comment:
This pattern appears in a lot of places, should we try to extract it or a part in order to enable reuse?

Also related to this, we have `KnownMethods` class which has some methods similar with this. Did you consider extracting the name check as a method there?

Reply:
`KnownMethods` operates on symbols, but I extracted it in extension methods.

I think anyway `ToString` has its cost as well, it probably iterates over the AST and calls the sub-nodes `ToString`s ... but at least if we want to change the method and improve it, we'll have a central place to do it